### DISCUSSION TITLE
Support multiple default values for params

### DISCUSSION THREAD

**Discussion by sgutwein** [knqyf263/pet#247] [Ideas]
Would be nice to have the option to define multiple default params. After executing the command it should be possible to choose between them.

E.g. `<stage=dev;qa;prod>` or something similar.

**Comment by smartmic** [comment]
+1 and (or alternatively) it would be also great if there is the option to show a short description/hint for every parameter.

**Comment by jkldrr** [comment]
+1

**Comment by jaroslawhartman** [comment]
It would be also something absolutely fantastic to build this list from a result of another script, for example
```
[[snippets]]

description = "Activate gcloud configuration"
command  =  "gcloud config configurations activate <param_command=gcloud config configurations list>"
```

Then in execution window a small list to select out of the `param_commands`.

Best regards,
Jarek

**Comment by hitzhangjie** [comment]
hey, guys. I support this in PR #203

**Comment by RamiAwar** [comment]
Only concern with this I'd say is the choice of the separator. `;` is used in bash to separate commands, `|` is used a LOT in different commands so that's definitely a no-go.

We just need to find the best character (least used, unrelated to bash) to be used as a separator. @hitzhangjie @sgutwein

**Comment by RamiAwar** [comment]
Saw this was also in #239 and #229

**Comment by jaroslawhartman** [comment]
Options for discussioin:

1. A least-used, single character. Arguably, there will be always a chance that chosen character will be used by somebody
2. A pair of tags, surrounding the option, e.g. `<param=[option1][option2]>` , perhaps with some escaping.

[Code suggestion](https://go.dev/play/p/vtGPhrzNj1a) 

3. Sed-like approach, similarly to its `s` command, e.g. `s|aaa|bbb|` or `s/aaa/bbb/` where it's up to the user to define separator.

Here:

* **Variant 1**

Use any character which appears after `param=`, for example:

```
<param=|option1|option2|option3>
<param=#option1#option2#option3#>
<param=!option1!option2!option3>
<param==option1=option2>
<param==option1>
```

* **Variant 1B**

Use any character which appears after `param=`, **as long as it's same at end** , for example:

```
<param=|option1|option2|option3|>
```

but...
```
<param=|option1|option2|option3>
```
means that `param=|option1|option2|option3`

Big benefit is that for single-parameter we keep nice fallback:
```
<param=option>
```
means simply `param=option`

* **Variant 2** - 

Separator defined as the first char after opening `<`

```
<|param|option1|option2>
<#param#option1#option2#>
```

I personally like Option 3 as it leaves to the user what char will be used, which effectively removes possible incompatibilities with user's needs. Personal preference, but Variant 1B looks a _best_ for me.

Cheers

**Reply by RamiAwar** [reply]
Great idea @jaroslawhartman , really like the flexible option as well! I think code wise it won't be too complex and still understandable. And user experience wise, this is a more 'advanced' feature, so the fact that they will have to read about in the docs is fine.

**Comment by brendan-stephens** [comment]
If the separator encapsulates the variables, it will also allow support for multiple word variables:
`<param=[John Doe][Jane Doe]>`

**Reply by RamiAwar** [reply]
This made me think of using two separators, like _|, that would make it easy to match and very unlikely to collide.

What do you think of something like this?
@jaroslawhartman @brendan-stephens 
<img width="282" alt="image" src="https://github.com/knqyf263/pet/assets/8523191/a07fc61b-20e9-4a6f-9baa-97048f45c391">

**Comment by jaroslawhartman** [comment]
As always - more unusual patter, less likely it will be used in text defined by the user. 

Pipes are popular in all types of shell but in combination with _ char - I don't recall I've ever used it in combination as you suggested.

**Comment by RamiAwar** [comment]
All that remains to support this is some nice dropdown/other for selecting those multiple values then. Interested in picking this one up yourself or should I @jaroslawhartman 😃 ? You've already done something similar

**Comment by RamiAwar** [comment]
This is now supported 🚀 🚀 🚀 ! @sgutwein @smartmic @jkldrr @hitzhangjie @brendan-stephens 

Instructions on how to use it are in the docs under Parameters.

https://github.com/knqyf263/pet/releases/tag/v0.7.0
